rustls/src/tls12/mod.rs

```diff
@@ -1,14 +1,14 @@
 use crate::cipher::{MessageDecrypter, MessageEncrypter};
 use crate::conn::{CommonState, ConnectionRandoms, Side};
 use crate::enums::{CipherSuite, SignatureScheme};
-use crate::kx;
 use crate::msgs::codec::{Codec, Reader};
-use crate::msgs::enums::{AlertDescription, ContentType};
+use crate::msgs::enums::AlertDescription;
 use crate::msgs::handshake::KeyExchangeAlgorithm;
 use crate::suites::{BulkAlgorithm, CipherSuiteCommon, SupportedCipherSuite};
 #[cfg(feature = "secret_extraction")]
 use crate::suites::{ConnectionTrafficSecrets, PartiallyExtractedSecrets};
 use crate::Error;
+use crate::{kx, InvalidMessage};
```

Comment:
Style nit: I would prefer to live `crate::kx` as is and import `Error` and `InvalidMessage` both directly from `crate::error`.